### PR TITLE
Extend merge functionality between mta.yaml and mta extension descriptor

### PR DESCRIPTION
**Pull request by tal-sapan**
### Checklist
- [X] Code compiles correctly
- [X] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [X] Formatting and linting run locally successfully
- [X] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: mta/ext.go [hunk 1]
```diff
@@ -5,38 +5,194 @@ import (
...
+
+// extendIncludes extends Slice Includes with elements of mta extension Slice Includes
+func extendIncludes(m *[]Includes, ext []Includes) error {
 	if ext != nil {
-		for key, value := range *ext {
-			(*m)[key] = value
+		if *m == nil {
+			*m = []Includes{}
 		}
+		*m = append(*m, ext...)
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply mta/ext.go]: Yes, the spec doesn't say it isn't allowed. Though in general we don't actually support include (it needs further processing - it points to a file which should either be packaged or embedded in the mtad.yaml, and it contains further properties that we might want to handle somehow). I'll keep this merge of includes here but won't merge other includes n the mta.yaml after talking to Natalia about it.

File: mta/ext_test.go [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
 
 import (
-	"os"
-	"path/filepath"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
+	"os"
+	"path/filepath"
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply mta/ext_test.go]: Done

File: mta/ext.go [hunk 3]
```diff
@@ -5,38 +5,194 @@ import (
...
+	return nil
+}
+
+// extendBool extends bool with element of mta extension bool
+func extendBool(m *bool, ext *bool) error {
+	if ext != nil {
+		*m = *ext
+	}
+
+	return nil
```
[the remark this answers is not in the document]
Comment (tal-sapan) [reply mta/ext.go]: Removed

File: mta/ext.go [hunk 4]
```diff
@@ -5,38 +5,300 @@ import (
...
+func (v *chainError) extendIncludes(m *[]Includes, ext []Includes, msg string, args ...interface{}) *chainError {
+	if v.err != nil {
+		return v
+	}
+	extendIncludes(m, ext)
+	return v
+}
+
+func (v *chainError) extendBool(m *bool, ext *bool, msg string, args ...interface{}) *chainError {
+	if v.err != nil {
```
Comment (allaVolkov) [inline mta/ext.go]: maybe shorter:
if v.err==nill{
   extendBool(m, ext)
}
return v

**Comment by tal-sapan** [comment]
Now merging:
* root parameters
* modules build-parameters
* modules parameters
* modules properties
* modules includes
* modules requires properties and parameters
* modules hooks parameters
* modules hooks requires properties and parameters
* resources properties
* resources parameters
* resources requires properties and parameters
* resources active

Maps are merged recursively (extended).
Trying to overwrite a non-overwritableble property or parameter is an error (overwritable is true by default). 
Trying to extend module, resource, provides, requires, hook etc which doesn't exist in the mta.yaml is an error.

**Pull request merged**